src/main/java/com/exactpro/th2/mstore/ProtoUtil.java

```diff
@@ -43,12 +45,17 @@ public static MessageToStore toCradleMessage(Message protoMessage) {
     public static MessageToStore toCradleMessage(RawMessage protoRawMessage) {
         RawMessageMetadata metadata = protoRawMessage.getMetadata();
         MessageID messageID = metadata.getId();
-        return new MessageToStoreBuilder()
+        MessageToStoreBuilder builder = new MessageToStoreBuilder()
                 .streamName(messageID.getConnectionId().getSessionAlias())
                 .content(protoRawMessage.toByteArray())
                 .timestamp(toInstant(metadata.getTimestamp()))
                 .direction(toCradleDirection(messageID.getDirection()))
-                .index(messageID.getSequence())
-                .build();
+                .index(messageID.getSequence());
+        metadata.getPropertiesMap().forEach((key, value) -> {
+            if (isNotBlank(key) || isNotBlank(value)) {
```

Comment:
Maybe `&&`? Both key and value shouldn't be blank?

Reply:
I think user or component can send empty key or value and we shouldn't drop data in this case

Reply:
OK, I see.